File: programs/benchmark/bubble_sort.go
/* Bubble sort. */

package main

/*
 * Performs in-place bubble sort on a list of integers
 *
 * @param list - Integer slice to sort
 * @param size - Size of list
 */
func sort(list []int, size int) {
	var swapped bool = true
 	
 	for swapped {
 		swapped = false

 		for i := 1; i < size; i++ {
 			if list[i - 1] > list[i] {
 				var tmp int = list[i - 1]
	 			list[i - 1] = list[i]
	 			list[i] = tmp

	 			swapped = true
 			}
 		}
 	}
}

// Seed for pseudo-random number generation. 
var seed int = 518
// Stores the last pseudo-random number generated.
var last int = seed

/* Generates a pseudo-random number between 0 and 10 inclusive. */
func rand() int {
	last = (7 * last + 9) % 11
	return last
}

func main() {
	var list []int
	var size int = 5786

	println("Generating list of", size, "random integers to sort...")

	for i := 0; i < size; i++ {
		list = append(list, rand())
	}

	println("Sorting the list of numbers...")

	sort(list, size)

	println("Sorted!")
}

File: programs/benchmark/bubble_sort.py
# Bubble sort.

def sort(l):
	"""
	Performs in-place bubble sort on a list of integers

	@param list - Integer slice to sort
	"""
	swapped = True
 	
 	while swapped:
 		swapped = False

 		for i in range(1, len(l)):
 			if l[i - 1] > l[i]:
 				tmp = l[i - 1]
	 			l[i - 1] = l[i]
	 			l[i] = tmp

	 			swapped = True

# Seed for pseudo-random number generation. 
seed = 518
# Stores the last pseudo-random number generated.
last = seed

# Generates a pseudo-random number between 0 and 10 inclusive.
def rand():
	global last
	last = (7 * last + 9) % 11
	return last

# Runs for about 4 sec. on Mimi.
def main():
	size = 5786

	print "Generating list of", size, "random integers to sort..."

	l = [rand() for _ in range(size)]

	print "Sorting the list of numbers..."

	sort(l)

	print "Sorted!"

if __name__ == '__main__':
	main()
